src/lib/context/getGlobalElements.ts

```diff
@@ -1,34 +1,33 @@
 import { GetServerSidePropsContext, GetStaticPropsContext } from 'next'
-import { transformBannerDataService } from '@/templates/globals/banner/dataService'
+import { formatter } from '@/data/queries/banners'
 import { drupalClient } from '@/lib/utils/drupalClient'
-import { LayoutProps } from '@/templates/globals/wrapper'
-
-type GlobalElements = LayoutProps
 
 // This is a helper function to fetch global elements for layout.
 // This is going to be run for every pages on build.
 // To make this fast, you could cache the results example on Redis.
+const nonSlugRoute = `${process.env.NEXT_PUBLIC_DRUPAL_BASE_URL}/jsonapi/banner-alerts?item-path=/`
 
 export async function getGlobalElements(
   context: GetStaticPropsContext | GetServerSidePropsContext
-): Promise<GlobalElements> {
+) {
   // global context
   const slug = await drupalClient.getPathFromContext(context)
   const path = await drupalClient.translatePathFromContext(context)
   const route = `${path?.jsonapi?.entryPoint}`
 
   let bannerPath = `${route}/banner-alerts?item-path=${slug}`
+
   if (slug.includes('home')) {
-    bannerPath = `${process.env.NEXT_PUBLIC_DRUPAL_BASE_URL}/jsonapi/banner-alerts?item-path=/`
+    bannerPath = `${nonSlugRoute}`
   }
 
   const requestBanner = await drupalClient.fetch(`${bannerPath}`)
   const bannerData = drupalClient.deserialize(await requestBanner.json())
-  const banner = await transformBannerDataService({ bannerData })
+  const banner = formatter({ bannerData } as any)
 
```

Comment:
Banner data formatter calls `src/data/queries/banners.ts`

Reply:
I've been struggling with identifying the correct type for `bannerData` so I resolved to using any for now